### PR TITLE
feat: embed: remove "Preview" tab and instead split code/preview layout 50/50 vertically

### LINKED ISSUE #15490: [CAL-3947] embed: remove "Preview" tab and instead split code/preview layout 50/50 vertically
before:

image

after:

image

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3947](https://linear.app/calcom/issue/CAL-3947/embed-remove-preview-tab-and-instead-split-codepreview-layout-5050)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

- Fixes #15490 
- Fixes CAL-3947 

There was some link issue that is why it is showing not found page. Should work fine with localhost

Screenshot 2024-06-19 184004

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

### CODE REVIEW

File: packages/features/embed/Embed.tsx [hunk 1]
```diff
@@ -1020,40 +1020,54 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
             </div>
           )}
         </div>
-        <div className="flex w-2/3 flex-col px-8 pt-8">
+        <div className="flow-y-scroll flex h-[95vh] w-2/3 flex-col px-8 pt-8">
```
Comment (hariombalhara) [inline packages/features/embed/Embed.tsx:1023]: 'flow-y-scroll' seems like a typo.

File: packages/features/embed/Embed.tsx [hunk 2]
```diff
@@ -1020,40 +1020,54 @@ const EmbedTypeCodeAndPreviewDialogContent = ({
...
+                          embedType={embedType}
+                          calLink={calLink}
+                          previewState={previewState}
+                          ref={iframeRef}
+                        />
+                      )}
+                    </div>
+                    <div className="flex-1">
+                      {previewTab && (
+                        <previewTab.Component
```
Comment (hariombalhara) [inline packages/features/embed/Embed.tsx:1062]: This is causing a preview tab per other tab. So, 2 preview tabs. We should have only one preview tab as there is a communication that happens with iframe and having 2 different iframes would have different results for different tabs.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Found an issue, left a comment.

File: packages/features/embed/Embed.tsx (hunk 1 above)
Comment (Amit91848) [reply packages/features/embed/Embed.tsx]: Yeah removed it

File: packages/features/embed/Embed.tsx (hunk 2 above)
Comment (Amit91848) [reply packages/features/embed/Embed.tsx]: I didn't realise that, made some changes please review again

**Comment by Amit91848** [comment]
Also iframe is throwing 404 on localhost, how do you enable it?

**Review by anikdhabal — COMMENTED** [review]
Make sure this scrollbar is not visible.
Screenshot 2024-07-02 081109


I think this is visible due to the changing height from 90 to 95vh

**Review by hariombalhara — DISMISSED** [review]
LGTM now !!

**Comment by hariombalhara** [comment]
I think we should think about what to do about the problem that @anikdhabal mentioned. The solution here I believe is to finalize what is the minimum screen size we want to support. Due to moving to column layout, we are in need of more height which is usually not that much as compared to width on big screens.

Also, I think we should make the border rounded for iframe as well image

**Comment by hariombalhara** [comment]
Regarding 404, I think you should just run `yarn dev` in embed-core
https://github.com/Amit91848/cal.com/blob/37f6ff660b06cbba3c32dbcda9be13a78507d92b/packages/embeds/embed-core/README.md#L16

**Comment by Amit91848** [comment]
> I think we should think about what to do about the problem that @anikdhabal mentioned. The solution here I believe is to finalize what is the minimum screen size we want to support. Due to moving to column layout, we are in need of more height which is usually not that much as compared to width on big screens.


> Also, I think we should make the border rounded for iframe as well image

**Review by anikdhabal — DISMISSED** [review]
Hey @Amit91848, you need to fix the test. I think you need to remove the function `goToPreviewTab(page)` and find a way to handle the function `expectToContainValidPreviewIframe()`

**Comment by hariombalhara** [comment]
@Amit91848 There are 2 tests failing related to Email Embed. Please check

**Comment by Amit91848** [comment]
All tests are passing now @hariombalhara .

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**